### PR TITLE
Add support for MyListener call getting updates to service TXT records (2nd attempt)

### PR DESCRIPTION
**Pull request by mattsaxon**
At the moment, the implementation supports notification to the ServiceListener class for additions and removals of service, but for service updates to the TXT record, the client must poll the ServiceInfo class. This draft PR provides a mechanism to have a callback on the ServiceListener class be invoked when the TXT record changes.

This resolves 
https://github.com/jstasiak/python-zeroconf/issues/165

and I think also
https://github.com/jstasiak/python-zeroconf/issues/98

I've also uncovered a couple of possible bugs/features in the testing for this, which I will raise as separate issues.

### CODE REVIEW

File: test_zeroconf.py [hunk 1]
```diff
@@ -968,3 +1074,8 @@ def send(out, addr=r._MDNS_ADDR, port=r._MDNS_PORT):
         assert service_removed.is_set()
         browser.cancel()
         zeroconf_browser.close()
+
+# Support for running outside of test tool
+# test = ListenerTest()
+# test.test_integration_with_listener_class()
+# test.test_integration_with_listener_class_with_update()
```
Comment (jstasiak) [inline test_zeroconf.py]: This is out of scope of this patch, I'd remove.

File: test_zeroconf.py [hunk 2]
```diff
@@ -807,6 +807,112 @@ def remove_service(self, zeroconf, type, name):
...
+            assert info is not None
+            assert info.properties[b'prop_blank'] == properties['prop_blank']
+            # End material test addition
+
+            zeroconf_registrar.unregister_service(info_service)
+            service_removed.wait(1)
+            assert service_removed.is_set()
+
+        except AssertionError:
+            raise
```
Comment (jstasiak) [inline test_zeroconf.py]: This `except` block can be removed, it doesn't seem to be doing anything.

File: test_zeroconf.py [hunk 3]
```diff
@@ -807,6 +807,112 @@ def remove_service(self, zeroconf, type, name):
...
+            assert info.properties[b'prop_string'] == properties['prop_string']
+            assert info.properties[b'prop_float'] is False
+            assert info.properties[b'prop_blank'] == properties['prop_blank']
+            assert info.properties[b'prop_true'] is True
+            assert info.properties[b'prop_false'] is False
+
+            info = zeroconf_browser.get_service_info(subtype, registration_name)
+            assert info is not None
+            assert info.properties[b'prop_none'] is False
+
```
Comment (jstasiak) [inline test_zeroconf.py]: Can you remove all the unnecessary (already verified in `test_integration_with_listener_class `) assertions above? Or actually maybe move the new code to the `test_integration_with_listener_class` method so that we don't have that much code repeated.

File: .travis.yml [hunk 4]
```diff
@@ -16,6 +16,8 @@ script:
     - make test_coverage
     - flake8 --version
     - make flake8
-    - if [[ "${TRAVIS_PYTHON_VERSION}" != "pypy"* ]] ; then make mypy ; fi
```
[the remark this answers is not in the document]
Comment (mattsaxon) [reply .travis.yml]: the 3.4 build was failing because mypy no longer supports it, so this change is to remove 3.4. I couldn't find the correct syntax for doing ORs in Travis, so I resorted to 3 positive IFs.

File: test_zeroconf.py (hunk 3 above)
Comment (mattsaxon) [reply test_zeroconf.py]: Yes, I will refactor as you suggest

File: test_zeroconf.py (hunk 2 above)
Comment (mattsaxon) [reply test_zeroconf.py]: Agreed, missed in my review!

File: test_zeroconf.py (hunk 1 above)
Comment (mattsaxon) [reply test_zeroconf.py]: OK

File: zeroconf.py [hunk 5]
```diff
@@ -1855,6 +1867,24 @@ def register_service(
...
+        self._broadcast_service(info)
+
+    def update_service(
+        self, info: ServiceInfo
+    ) -> None:
+        """Registers service information to the network with a default TTL.
+        Zeroconf will then respond to requests for information for that
+        service."""
+
+        assert(self.services[info.name.lower()] is not None)
```
[the remark this answers is not in the document]
Comment (mattsaxon) [reply zeroconf.py]: Will do

File: .travis.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jstasiak) [reply .travis.yml]: That's fair, I removed 3.4 in master just now so if you rebase your branch this particular change will no longer be necessary.

**Comment by mattsaxon** [comment]
@jstasiak, I've made the amendment from your code review. Can you take another look please?

**Comment by jstasiak** [comment]
Very nice, thank you! Just merged this.

**Comment by mattsaxon** [comment]
Thanks and no problem. Should #165 and #98 be closed too, or will you wait till next release?

**Comment by jstasiak** [comment]
I think let's close those once it's released (I'll probably come around to doing it in the next few days).

**Comment by zougloub** [comment]
Thank you @mattsaxon for this work.
